### PR TITLE
Add explicit Weights and Biases support

### PR DESCRIPTION
**Pull request by Clemspace**
changes in readme.md and default examples/7b.yml file for more explicit wandb variable use

### CODE REVIEW

**Comment by Clemspace** [comment]
Also a more explcit name for lora weights, like we discussed! ;)

File: example/7B.yaml [hunk 1]
```diff
@@ -25,6 +25,12 @@ eval_freq: 100
 no_eval: False
 ckpt_freq: 100
 
-save_adapters: True  # save only trained LoRA adapters. Set to `False` to merge LoRA adapter into the base model and save full fine-tuned model
+ckpt_only_lora: True  # save only trained LoRA adapters. Set to `False` to merge LoRA adapter into the base model and save full fine-tuned model
```
Comment (patrickvonplaten) [inline example/7B.yaml:28]:
```suggestion
save_adapters: True  # save only trained LoRA adapters. Set to `False` to merge LoRA adapter into the base model and save full fine-tuned model
```

File: train.py [hunk 2]
```diff
@@ -310,14 +307,13 @@ def _train(
             (args.ckpt_freq > 0 and state.step % args.ckpt_freq == 0) or is_last_step
         ):
             checkpointer.save_checkpoint(
-                save_only_lora=args.save_adapters,
+                save_only_lora=args.ckpt_only_lora,
```
Comment (patrickvonplaten) [inline train.py:310]:
```suggestion
                save_only_lora=args.save_adapters,
```

File: README.md [hunk 3]
```diff
@@ -473,6 +473,27 @@ Assuming your `lora.safetensors` is saved under `$HOME/ultra_chat_test/checkpoin
...
+### Setting Up Weights and Biases
+
+To use Weights and Biases with `mistral-finetune`, follow these steps:
+
+1. **Install Weights and Biases:**
+
+   Make sure you have the `wandb` library installed. You can install it using pip:
+
+   ```bash
+   pip install wandb
```
Comment (patrickvonplaten) [inline README.md:489]:
```suggestion
   pip install wandb
   ```
```

Comment (patrickvonplaten) [inline README.md:489]: Some backticks are missing here: ```

File: example/7B.yaml [hunk 4]
```diff
@@ -28,3 +28,9 @@ ckpt_freq: 100
 save_adapters: True  # save only trained LoRA adapters. Set to `False` to merge LoRA adapter into the base model and save full fine-tuned model
 
 run_dir: ""  # Fill
+
+wandb:
```
Comment (patrickvonplaten) [inline example/7B.yaml:32]: nice!

File: finetune/monitoring/metrics_logger.py [hunk 5]
```diff
@@ -14,7 +14,6 @@
```
Comment (patrickvonplaten) [inline finetune/monitoring/metrics_logger.py:14]: Can we remove all changes to this file?

File: train.py [hunk 6]
```diff
@@ -50,12 +50,10 @@
 
 logger = logging.getLogger("train")
 
-
```
Comment (patrickvonplaten) [inline train.py:53]: Can we remove all changes to this file except correcting the gra d type?

**Pull request merged**